specs/CustomStoragePartition.md

```diff
@@ -0,0 +1,208 @@
+  /// This feature requires enabling 2 experimental browser features to work properly.
+  /// These features will be enabled by default in the future.
+  /// Before these features are enabled by default, please enable them by ensuring
+  /// `--enable-features=ThirdPartyStoragePartitioning,PartitionedCookies` is set in
+  /// `AdditionalBrowserArguments` in `CoreWebView2EnvironmentOptions` used to create
+  /// CoreWebView2Environment. If these features are not enabled, all data are treated
+  /// as unpartitioned and stored in the global default location for the profile.
+  /// When it is set, the page in the WebView will act as if the page were hosted in a
+  /// top level site uniquely associated with the `partitionId` and have a separate 
+  /// storage partition as described in https://developer.chrome.com/docs/privacy-sandbox/storage-partitioning/
+  /// and separete cookie partition as described in https://developer.chrome.com/docs/privacy-sandbox/chips/
+  /// with all cookies partitioned.
+  /// If `customStoragePartitionId` is nullptr or empty string, the
+  /// `CustomStoragePartitionId` will be reset and the page inside the WebView
+  /// will work normally with data treated as unpartitioned.
+  /// The `customStoragePartitionId` parameter is case sensitive. The default is
+  /// an empty string. There is no restriction on the length or what characters
+  /// can be used in partition id.
+  /// The change of the custom storage partition id will be applied to new 
+  /// page or iframe navigations and not impact existing pages and iframes.
+  /// To avoid accidentally use new partition id for pending navigations of old page
+  /// or iframe, it is recommended to create a new WebView for new partition instead
+  /// of changing partition. If you really have to change partition, it is
+  /// recommended to navigate to a blank page before setting the new partition
+  /// id and navigate to a page for the new partition.
+  /// 
+  /// As setting custom storage partition id does not change DOM security
+  /// model, developers should be very careful for WebViews with opener and
+  /// opened window relationship, especially when the pages in the WebViews
+  /// have same origin, like when the opened window is the same site or
+  /// about:blank. The pages in these WebViews can access each other’s DOM and
+  /// therefore can potentially access DOM storage and cookies in different
+  /// partition for the same site. It is recommended to set the same custom
+  /// storage partition id for these WebViews, unless there is an absolute need
+  /// to set different partition ids and only trusted code is hosted in them.
+  ///
+  [propput] HRESULT CustomStoragePartitionId([in] LPCWSTR customStoragePartitionId);
+}
+
+interface ICoreWebView2Profile8 : ICoreWebView2Profile7 {
+  /// Clears all DOM storage and cookies in the custom storage partition 
+  /// identified by the `customStoragePartitionId`.
```

Comment:
If I pass in null or empty, does it clear the default partition?

Reply:
The API actually will fail with E_INVALIDARG. Should add it: If `customStoragePartitionId` is null or an empty string, the API will fail with E_INVALIDARG.

Reply:
See above.